lib/pure/parsecfg.nim

```diff
@@ -295,252 +371,440 @@ proc getEscapedChar(c: var CfgParser, tok: var Token) =
+    rawGetTok(c)
 
 # ---------------- Configuration file related operations ----------------
 type
-  Config* = OrderedTableRef[string, <//>OrderedTableRef[string, string]]
+  Config* = OrderedTableRef[string, SectionItem]
+
+  SectionItem = tuple
+    sectionRelated: SectionRelated
+    keyValue: <//>OrderedTableRef[string, KeyValueItem]
+
+  KeyValueItem = tuple
+    value: string
+    keyValueRelated: KeyValueRelated
 
-proc newConfig*(): Config =
+proc newConfig*(): <//>Config =
   ## Creates a new configuration table.
   ## Useful when wanting to create a configuration file.
-  result = newOrderedTable[string, <//>OrderedTableRef[string, string]]()
+  result = newOrderedTable[string, SectionItem]()
 
-proc loadConfig*(stream: Stream, filename: string = "[stream]"): <//>Config =
+proc loadConfig*(stream: Stream, filename: string = "[stream]",
+                 commentSymbol = "#;"): <//>Config =
   ## Loads the specified configuration from stream into a new Config instance.
   ## `filename` parameter is only used for nicer error messages.
-  var dict = newOrderedTable[string, <//>OrderedTableRef[string, string]]()
+  ## `commentSymbol` default value is `"#;"`
+  var dict = newOrderedTable[string, SectionItem]()
   var curSection = "" ## Current section,
                       ## the default value of the current section is "",
                       ## which means that the current section is a common
   var p: CfgParser
   open(p, stream, filename)
+  p.commentSymbol = commentSymbol
   while true:
     var e = next(p)
     case e.kind
     of cfgEof:
       break
     of cfgSectionStart: # Only look for the first time the Section
+      var tp: SectionItem
+      var t = <//>OrderedTableRef[string, KeyValueItem]()
       curSection = e.section
+      tp.sectionRelated = e.sectionRelated
+      tp.keyValue = t
+      dict[curSection] = tp
     of cfgKeyValuePair:
-      var t = newOrderedTable[string, string]()
+      var tp: SectionItem
+      var t = <//>OrderedTableRef[string, KeyValueItem]()
       if dict.hasKey(curSection):
-        t = dict[curSection]
-      t[e.key] = e.value
-      dict[curSection] = t
+        tp = dict[curSection]
+        t = tp.keyValue
+      t.add(e.key, (e.value, e.keyValueRelated))
+      tp.keyValue = t
+      dict[curSection] = tp
     of cfgOption:
-      var c = newOrderedTable[string, string]()
+      var tp: SectionItem
+      var t = <//>OrderedTableRef[string, KeyValueItem]()
       if dict.hasKey(curSection):
-        c = dict[curSection]
-      c["--" & e.key] = e.value
-      dict[curSection] = c
+        tp = dict[curSection]
+        t = tp.keyValue
+      t[e.key] = (e.value, e.keyValueRelated)
+      tp.keyValue = t
+      dict[curSection] = tp
     of cfgError:
-      break
+      close(p)
+      raise newException(Exception, e.msg)
   close(p)
   result = dict
 
-proc loadConfig*(filename: string): <//>Config =
+proc loadConfig*(filename: string, commentSymbol = "#;"): <//>Config =
```

Comment:
`commentSymbols = {'#', ';'}`

Reply:
All of these optimizations are necessary, but for now My focus is on the inspection not passing.

Reply:
Was not about optimization, is more correctness of the code too, 
when something is a char then must be a `char`, Nim has a concrete type for that.

e.g. `commentSymbol = "##"` and :bomb: :boom:  
But `commentSymbol = {'#', '#'}` can not be.

Reply:
The main thing is that I now add the function of empty lines and comment lines to 'parsecfg', but 'nimble' checks that it doesn't read 'key' as desired when he calls' parsecfg 'to read' key ', so the checks are always wrong. It seems we need to modify 'nimble'.